packages/dht/src/dht/DhtNode.ts

```diff
@@ -0,0 +1,776 @@
+    }
+
+    public async joinDht(entryPointDescriptor: PeerDescriptor, doRandomJoin?: boolean): Promise<void> {
+        if (!this.started) {
+            throw new Error('Cannot join DHT before calling start() on DhtNode')
+        }
+        await this.peerDiscovery!.joinDht(entryPointDescriptor, doRandomJoin)
+    }
+
+    public async startRecursiveFind(idToFind: Uint8Array, findMode?: FindMode, excludedPeer?: PeerDescriptor): Promise<RecursiveFindResult> {
+        return this.recursiveFinder!.startRecursiveFind(idToFind, findMode, excludedPeer)
+    }
+
+    public async storeDataToDht(key: Uint8Array, data: Any): Promise<PeerDescriptor[]> {
+        return this.dataStore!.storeDataToDht(key, data)
+    }
+
+    public async getDataFromDht(idToFind: Uint8Array): Promise<RecursiveFindResult> {
+        return this.recursiveFinder!.startRecursiveFind(idToFind, FindMode.DATA)
+    }
+
+    public async deleteDataFromDht(idToDelete: Uint8Array): Promise<void> {
+        if (!this.stopped) {
+            return this.dataStore!.deleteDataFromDht(idToDelete)
+        }
+    }
+
+    public async findDataViaPeer(idToFind: Uint8Array, peer: PeerDescriptor): Promise<DataEntry[]> {
+        const target = new RemoteExternalApi(
+            this.ownPeerDescriptor!,
+            peer,
+            toProtoRpcClient(new ExternalApiServiceClient(this.rpcCommunicator!.getRpcClientTransport())),
+            this.config.serviceId
+        )
+        return await target.findData(idToFind)
+    }
+
+    public getRpcCommunicator(): RoutingRpcCommunicator {
+        return this.rpcCommunicator!
+    }
+
+    public getTransport(): ITransport {
+        return this.transportLayer!
+    }
+
+    public getPeerDescriptor(): PeerDescriptor {
+        return this.ownPeerDescriptor!
+    }
+
+    public getAllConnectionPeerDescriptors(): PeerDescriptor[] {
+        return Array.from(this.connections.values()).map((peer) => peer.getPeerDescriptor())
+    }
+
+    public getK(): number {
+        return this.config.numberOfNodesPerKBucket
+    }
+
+    public getKBucketPeers(): PeerDescriptor[] {
+        return this.bucket!.toArray().map((dhtPeer: DhtPeer) => dhtPeer.getPeerDescriptor())
+    }
+
+    public getOpenInternetPeerDescriptors(): PeerDescriptor[] {
+        return this.openInternetPeers!.getAllContacts().map((contact) => contact.getPeerDescriptor())
+    }
+
+    public getNumberOfConnections(): number {
+        return this.connections.size
+    }
+
+    public getNumberOfLocalLockedConnections(): number {
+        return this.connectionManager!.getNumberOfLocalLockedConnections()
+    }
+
+    public getNumberOfRemoteLockedConnections(): number {
+        return this.connectionManager!.getNumberOfRemoteLockedConnections()
+    }
+
+    public getNumberOfWeakLockedConnections(): number {
+        return this.connectionManager!.getNumberOfWeakLockedConnections()
+    }
+
+    public getNodeName(): string {
```

Comment:
Could return `string | undefined`. If we need defined value in some UI, we can fallback it there (works also if someone wants to have e.g. localized fallback value)